pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -475,9 +475,15 @@ func translateRoute(node *model.Proxy, in *networking.HTTPRoute,
 			d = features.DefaultRequestTimeout
 		}
 		action.Timeout = d
-		// Use deprecated value for now as the replacement MaxStreamDuration has some regressions.
-		// nolint: staticcheck
-		action.MaxGrpcTimeout = d
+		if node.Metadata.Generator == "grpc" {
```

Comment:
nit: If we are going to add these checks I feel like we should have some function abstracting this - how we identify its for grpc may change in the future, etc. Seems more clear to have `node.IsGrpc()` or similar